torch/functional.py

```diff
@@ -386,16 +390,16 @@ def meshgrid(*tensors):
             True
 
         """
-        return _meshgrid(*tensors)
+        return _meshgrid(*tensors, indexing=indexing)
 
 
-def _meshgrid(*tensors):
+def _meshgrid(*tensors, indexing: Optional[str]):
     if has_torch_function(tensors):
-        return handle_torch_function(meshgrid, tensors, *tensors)
+        return handle_torch_function(meshgrid, tensors, *tensors, indexing=indexing)
     if len(tensors) == 1 and isinstance(tensors[0], (list, tuple)):
         # the old interface of passing the operands as one list argument
         tensors = tensors[0]  # type: ignore[assignment]
-    return _VF.meshgrid(tensors)  # type: ignore[attr-defined]
+    return _VF.meshgrid(tensors, indexing=indexing)  # type: ignore[attr-defined]
```

Comment:
When someone calls meshgrid won't they call the functional and always call into the new signature from it using this approach?

Reply:
Yes, per above comment, do we need to ensure that calls without indexing still go to the old version?

Reply:
Yes, because otherwise programs that just call meshgrid will hit an FC issue and won't be able to run on older versions of PyTorch (even versions of PyTorch that are just hours old!)

Reply:
is this what you had in mind as a fix?